yt/frontends/ramses/data_structures.py

```diff
@@ -362,23 +365,28 @@ def retrieve_ghost_zones(self, ngz, fields, smoothed=False):
             )
             smoothed = False
 
+        _subset_with_gz = getattr(self, "_subset_with_gz", {})
+
         try:
-            new_subset = self._subset_with_gz
+            new_subset = _subset_with_gz[ngz]
             mylog.debug(
-                "Reusing previous subset with ghost zone for domain %s", self.domain_id
+                "Reusing previous subset with %s ghost zones for domain %s",
+                ngz,
+                self.domain_id,
             )
-        except AttributeError:
+        except KeyError:
             new_subset = RAMSESDomainSubset(
```

Comment:
Would this result in, for instance, having both the entire subset *and* the subset + GZ in memory?

Reply:
Yes; the reason I decided to store them in memory is that computing the ghost zones is (very) slow, as it grows roughly as `N log(N) (Ngz**3-8)`, where `N` is the number of octs in the subset.